### PR TITLE
Add the discount email notification for the vendor you're following

### PR DESCRIPTION
**Pull request by VolodymyrBushko**

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java [hunk 1]
```diff
@@ -67,4 +65,16 @@ public DiscountResponseVO update(Long id, DiscountRequestVO request) {
...
+        Long vendorId = discount.getVendor().getId();
+        List<String> subEmails = vendorService.findAllSubEmailsByVendorId(vendorId);
+
+        if (!subEmails.isEmpty()) {
+            mailSender.sendMail(
+                    discount.getTitle(),
+                    discount.getShortDescription(),
+                    subEmails.toArray(new String[0]));
+        }
+    }
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java]: move to separate service

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java [hunk 2]
```diff
@@ -71,4 +75,32 @@ private Vendor getVendorById(Long id) {
         return vendorRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException(Vendor.class, "id", id));
     }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void subscribe(Long vendorId, String userEmail) {
+        Vendor vendor = getVendorById(vendorId);
+        User user = userService.getUserByEmail(userEmail);
+
+        vendorValidator.checkDuplicateSubscriber(vendor, user);
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: remove validation check

File: src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java (hunk 1 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java]: done

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java (hunk 2 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: removed

File: src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java [hunk 3]
```diff
@@ -49,15 +53,15 @@ public VendorResponseVO findByTitle(String title) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public VendorResponseVO create(VendorRequestVO request) {
-        vendorEmailValidator.validate(request);
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java]: why did you rename back vendorEmailVaildator to vendorValidator?

File: src/main/java/com/exadel/discountwebapp/vendor/validator/VendorValidator.java [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 @Component
 @RequiredArgsConstructor
-public class VendorEmailValidator {
+public class VendorValidator {
```
Comment (dmitriychernov7) [inline src/main/java/com/exadel/discountwebapp/vendor/validator/VendorValidator.java]: why?

**Review by dmitriychernov7 — APPROVED** [review]
